packages/cli/src/api/formats/po.test.js

```diff
@@ -71,7 +90,61 @@ describe("pofile format", function() {
     })
 
     const filename = path.join("locale", "en", "messages.po")
-    expect(format.read(filename)).toMatchSnapshot()
+    const actual = format.read(filename)
+    mockFs.restore()
+    expect(actual).toMatchSnapshot()
+  })
+
+  it("should correct badly used comments", function() {
+    const po = PO.parse(`
+      #. First description
+      #. Second comment
+      #. Third comment
+      msgid "withMultipleDescriptions"
+      msgstr "Extra comments are separated from the first description line"
+
+      # Translator comment
+      #. Single description only
+      #. Second description?
+      msgid "withDescriptionAndComments"
+      msgstr "Second description joins translator comments"
+    `)
+
+    mockFs({
+      locale: {
+        en: {
+          "messages.po": po.toString()
+        }
+      }
+    })
+
+    const filename = path.join("locale", "en", "messages.po")
+    const actual = format.read(filename)
+    mockFs.restore()
+    expect(actual).toMatchSnapshot()
+  })
+
+  it("should throw away additional msgstr if present", function() {
+    const po = PO.parse(`
+      msgid "withMultipleTranslation"
+      msgstr[0] "This is just fine"
+      msgstr[1] "Throw away that one"
+    `)
+
+    mockFs({
+      locale: {
+        en: {
+          "messages.po": po.toString()
+        }
+      }
+    })
+
+    const restoreConsole = mockConsole()
```

Comment:
How is this module better than [`mockConsole`](https://github.com/lingui/js-lingui/blob/072afd8d3664dce514c0628f7e77eef207682ce6/packages/cli/src/mocks.js#L12) in `src/mocks.js`? I don't mind extra dependency, it's dev code after all, but I would rather stick to one way of mocking console, so it's consistent across the repo.

I'm not sure how it handles exceptions before `restoreConsole` is called. What happens if format.read raises an error, does it restore console somehow automatically?

Reply:
Oh, I haven't noticed there is already that function in place, sorry about that. I tried throwing an error in `format.read` and it works just fine. I suppose that Jest is not using console, but writing to `stdout/err` directly so it doesn't matter if you mock it. Either way, I've used your approach, I kinda like it more since you don't need to call restore manually :)